api/apiclient.go

```diff
@@ -591,14 +592,27 @@ func dialAPI(info *Info, opts0 DialOpts) (*dialResult, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	dialInfo, err := dialWebsocketMulti(ctx, info.Addrs, path, opts)
+	dialInfo, err := dialWebsocketMulti(ctx, shuffleAddresses(info.Addrs), path, opts)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	logger.Infof("connection established to %q", dialInfo.urlStr)
 	return dialInfo, nil
 }
 
+func init() {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
assuming you do move the code to worker/apicaller, please create a worker-specific rand.Rand rather than modifying global state